storm-core/src/jvm/org/apache/storm/daemon/StormCommon.java

```diff
@@ -0,0 +1,604 @@
+
+    public static final String SYSTEM_STREAM_ID = "__system";
+
+    public static final String EVENTLOGGER_COMPONENT_ID = "__eventlogger";
+    public static final String EVENTLOGGER_STREAM_ID = "__eventlog";
+
+    public static void startMetricsReporter(PreparableReporter report, Map conf) {
+        report.prepare(new MetricRegistry(), conf);
+        report.start();
+        LOG.info("Started statistics report plugin...");
+    }
+
+    public static void startMetricsReporters(Map conf) {
+        List<PreparableReporter> reporters = MetricsUtils.getPreparableReporters(conf);
+        for (PreparableReporter reporter : reporters) {
+            startMetricsReporter(reporter, conf);
+        }
+    }
+
+    public static String getTopologyNameById(String topologyId) {
+        String topologyName = null;
+        try {
+            topologyName = topologyIdToName(topologyId);
+        } catch (InvalidTopologyException e) {
+            LOG.error("Invalid topologyId=" + topologyId);
+        }
+        return topologyName;
+    }
+
+    /**
+     * Convert topologyId to topologyName. TopologyId = topoloygName-counter-timeStamp
+     *
+     * @param topologyId
+     * @return
+     */
+    public static String topologyIdToName(String topologyId) throws InvalidTopologyException {
+        String ret = null;
+        int index = topologyId.lastIndexOf('-');
+        if (index != -1 && index > 2) {
+            index = topologyId.lastIndexOf('-', index - 1);
+            if (index != -1 && index > 0)
+                ret = topologyId.substring(0, index);
+            else
+                throw new InvalidTopologyException(topologyId + " is not a valid topologyId");
+        } else
+            throw new InvalidTopologyException(topologyId + " is not a valid topologyId");
+        return ret;
+    }
+
+    public static String getStormId(IStormClusterState stormClusterState, final String topologyName) {
+        List<String> activeTopologys = stormClusterState.activeStorms();
+        IPredicate pred = new IPredicate<String>() {
+            @Override
+            public boolean test(String obj) {
+                return obj != null ? getTopologyNameById(obj).equals(topologyName) : false;
+            }
+        };
+        return Utils.findOne(pred, activeTopologys);
+    }
+
+    public static Map<String, StormBase> topologyBases(IStormClusterState stormClusterState) {
+        return _instance.topologyBasesImpl(stormClusterState);
+    }
+
+    protected Map<String, StormBase> topologyBasesImpl(IStormClusterState stormClusterState) {
+        List<String> activeTopologys = stormClusterState.activeStorms();
+        Map<String, StormBase> stormBases = new HashMap<String, StormBase>();
+        if (activeTopologys != null) {
+            for (String topologyId : activeTopologys) {
+                StormBase base = stormClusterState.stormBase(topologyId, null);
+                if (base != null) {
```

Comment:
The original code didn't check for null here.  Is this needed?  Extra checks that fix things up like this make me nervous that be broke something somewhere and we are using this to work around it.